src/EvoSC.Common/Interfaces/Services/IPlayerManagerService.cs

```diff
@@ -56,6 +56,11 @@ public interface IPlayerManagerService
     /// <param name="player">Information about the player to get.</param>
     /// <returns>Returns an IOnlinePlayer instance of the player.</returns>
     public Task<IOnlinePlayer> GetOnlinePlayerAsync(IPlayer player);
+
+    /// <summary>
+    /// Get a player by their server ID.
+    /// </summary>
+    public IOnlinePlayer? GetOnlinePlayerByServerIdAsync(int playerServerId);
```

Comment:
Maybe note that this is O(N)? Because GetOnlinePlayerAsync is O(1) unless there is a cache miss.

It makes this method way slower than GetOnlinePlayerAsync on average. You could also create a map for these ID's, but I'm not sure if it will really be used that much for it to be worth the effort.

Reply:
The method is mostly useful for the few cases where you have a player server ID from XML-RPC. For the most part we could rely on Account ID.